File: src/main/java/pw/mihou/amelia/io/rome/ItemWrapper.java
package pw.mihou.amelia.io.rome;

import com.apptastic.rssreader.Item;
import pw.mihou.amelia.Amelia;

import java.text.ParseException;
import java.util.Date;
import java.util.Optional;

public class ItemWrapper {

    String title;
    Date date;
    String author;
    String link;
    String description;

    public ItemWrapper(Item item) {
        this.title = item.getTitle().orElse("");
        this.description = item.getDescription().orElse("");
        this.author = item.getAuthor().orElse("");
        this.link = item.getLink().orElse("");
        this.date = item.getPubDate().map(source -> {
            try {
                return Amelia.formatter.parse(source);
            } catch (ParseException e) {
                Amelia.log.error("Amelia wasn't able to parse the date {}, exception: {}", source, e.getMessage());
                return null;
            }
        }).orElse(null);
    }

    public ItemWrapper(String title, Date date, String author, String link, String description) {
        this.title = title;
        this.date = date;
        this.author = author;
        this.link = link;
        this.description = description;
    }

    public Optional<Date> getPubDate(){
        return Optional.ofNullable(date);
    }

    public String getTitle(){
        return title;
    }

    public String getAuthor(){
        return Optional.ofNullable(author).orElse("");
    }

    public boolean valid(){
        return link != null && title != null && !link.isEmpty() && !link.isBlank() && !title.isEmpty() && !title.isBlank();
    }

    public String getLink(){
        return link;
    }

    public String getDescription(){
        return description;
    }

}

File: src/main/java/pw/mihou/amelia/io/rome/ItemWrapper.kt
package pw.mihou.amelia.io.rome

import com.apptastic.rssreader.Item
import pw.mihou.amelia.Amelia.formatter
import pw.mihou.amelia.logger
import java.text.ParseException
import java.util.*

class ItemWrapper(item: Item) {

    val title: String
    val date: Date?
    val author: String
    val link: String
    val category: String?
    private val description: String

    init {
        title = item.title.orElse("")
        description = item.description.orElse("")
        author = item.author.orElse("")
        link = item.link.orElse("")
        date = item.pubDate.map { source: String ->
            try {
                return@map formatter.parse(source)
            } catch (e: ParseException) {
                logger.error("Amelia wasn't able to parse the date {}, exception: {}", source, e.message)
                return@map null
            }
        }.orElse(null)
        category = item.category.orElse(null)
    }

    fun valid(): Boolean {
        return link.isNotEmpty() && link.isNotBlank() && title.isNotEmpty() && title.isNotBlank()
    }

    override fun toString(): String {
        return "{title=$title, date=$date, author=$author, link=$link, description=$description, category=$category}"
    }
}